<launch>


  <!-- Nodelet manager for this pipeline -->
  <node pkg="nodelet" type="nodelet" args="manager"
        name="record_player_manager" output="screen"/>

  <!-- The depth image is already rectified and registered
       to the camera optical frame, but stored in mm; convert it
       to meters -->
  <node pkg="nodelet" type="nodelet" name="metric_rect" 
        args="load depth_image_proc/convert_metric record_player_manager --no-bond">
    <remap from="image_raw" to="/sim_image_depth/image_rect"/>
    <remap from="image" to="/sim_image_depth/image"/>
  </node>

  <!-- Convert it into a point cloud -->
  <node pkg="nodelet" type="nodelet" name="cloudify"
        args="load depth_image_proc/point_cloud_xyzrgb record_player_manager --no-bond">
    <remap from="depth_registered/image_rect" to="/sim_image_depth/image"/>
    <remap from="depth_registered/points" to="/sim_image_depth/points"/>
    <remap from="rgb/image_rect_color" to="/sim_image_rgb/image_rect"/>
    <remap from="rgb/camera_info" to="/sim_image_rgb/camera_info"/>
  </node>
</launch>
